Add default organization

tw4l opened the pull request:
Related to #455

Connected to #457 

- Add `default` switch to `Archive` (org) model
- Set default org name via values.yaml
- Add check to ensure only one org with default org name exists
- Stop creating new orgs for new users
- Add new API endpoints for creating and renaming orgs
- Make `Archive.name` unique via index
- Make archive-less invites invite user to default org

tw4l committed:
Add default organization

- Set default org name via values.yaml
- Add check to ensure only one org with default org name exists
- Stop creating new archives for new non-superusers

ikreymer commented:
Nice! One change: currently, the superadmin gets added as a user to the org / will be listed as an 'Owner'. To avoid confusion, we should not actually add the superadmin as user, as the superadmin will have access anyway (and can't be removed).

tw4l committed:
Create default org separately from superadmin

tw4l commented:
@ikreymer Good point! I separated out the default org creation from superadmin creation and switched it so that a new org isn't created for new superadmins. Should be good now!

tw4l committed:
Add additional features

- Add default switch to Archive model
- Add mutex to ensure only one default org is created
- Add index to make org names unique
- Use default flag to retrieve default org
- Add archive rename API endpoint

tw4l committed:
Add default org and org rename tests

tw4l committed:
Add missing await

tw4l committed:
Add API endpoint to create new org and tests

tw4l committed:
Init index in create_default_org

tw4l committed:
Wait for db to be ready to create index

tw4l committed:
Log waiting for db connection

tw4l committed:
Add default org to archive-less invites

tw4l committed:
Add second fallback to account for all code paths

tw4l committed:
Make sure invited user has role in default org

tw4l committed:
Give superadmin invited users owner role

tw4l committed:
Don't create new org for invited users

tw4l committed:
Rename default org from chart value if changed
